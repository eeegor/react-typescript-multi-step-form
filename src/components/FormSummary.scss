.table {
	font-size: 1rem;
	margin-bottom: 1rem;
	text-align: left;
}

.table__row {
	display: grid;
	grid-template-columns: 1fr;
	padding: 0.5rem 1rem;

	@media (min-width: 767px) {
		grid-template-columns: 1fr 3fr;
	}

	&:nth-child(odd) {
		background: #fafafa;
	}

	&:nth-child(even) {
		background: #fff;
	}
}

.table__row__label {
	opacity: 0.5;
}

.table__row__value {
	color: #333;
}
