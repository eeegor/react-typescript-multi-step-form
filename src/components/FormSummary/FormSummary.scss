@import '../../variables';

.table {
	text-align: left;
	margin: 2rem auto 2.5rem;
}

.table__row {
	display: grid;
	grid-template-columns: 1fr 3fr auto;
	padding: 1rem 0;
	border-top: 1px solid #eee;
	border-bottom: 1px solid #eee;
	margin-bottom: -1px;

	@media (min-width: 767px) {
		grid-template-columns: 1fr 3fr auto;
	}
}

.table__row__label {
	opacity: 0.5;
}

.table__row__value {
	color: #333;
}
