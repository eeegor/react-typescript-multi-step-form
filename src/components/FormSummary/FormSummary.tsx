/**
 * Component FormSummary
 */

import * as React from 'react';
import * as classnames from 'classnames';
import { Label, Button } from '..';
import './FormSummary.scss';

export interface Props {
	className?: string;
	formFields: object;
	label?: string | JSX.Element;
	gotoStep?: (step: number) => void;
}

export class FormSummary extends React.Component<Props> {
	render(): JSX.Element {
		const { className, label, formFields, gotoStep } = this.props;

		return (
			<div className={classnames('form-summary', className)}>
				{label && <Label className="form-summary__label" label={label} />}
				<div className="table">
					{Object.keys(formFields).map((formField, index) => (
						<div className="table__row" key={formField}>
							<div className="table__row__column">
								<div className="table__row__label">{formField}</div>
								<div className="table__row__value">{formFields[formField]}</div>
							</div>
							<div className="table__row__edit">
								<Button
									size="small"
									onClick={() => gotoStep && gotoStep(index + 1)}
								>
									Edit
								</Button>
							</div>
						</div>
					))}
				</div>
			</div>
		);
	}
}
