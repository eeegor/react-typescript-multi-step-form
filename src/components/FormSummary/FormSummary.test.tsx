import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { FormSummary } from './FormSummary';
import { mount } from 'enzyme';

describe('FormSummary', () => {
	const callback = jest.fn();

	it('renders markup correctly', () => {
		const wrapper = <FormSummary gotoStep={callback} formFields={{ name: '' }} />;
		const tree = renderer.create(wrapper).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('click event works', () => {
		const wrapper = mount(<FormSummary gotoStep={callback} formFields={{ name: '' }} />);
		wrapper
			.find('.form-summary__edit')
			.first()
			.simulate('click');
		expect(callback).toHaveBeenCalledTimes(1);
	});

	it('should adopt a label', () => {
		const wrapper = mount(<FormSummary label="hello-label" gotoStep={callback} formFields={{ name: '' }} />);
		expect(wrapper.find('.label')).toHaveLength(1);
	});
});
