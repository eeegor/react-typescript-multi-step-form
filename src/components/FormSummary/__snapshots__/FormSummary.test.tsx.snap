// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSummary renders markup correctly 1`] = `
<div
  className="form-summary"
>
  <div
    className="table"
  >
    <div
      className="table__row"
    >
      <div
        className="table__row__column"
      >
        <div
          className="table__row__label"
        >
          name
        </div>
        <div
          className="table__row__value"
        >
          
        </div>
      </div>
      <div
        className="table__row__edit"
      >
        <button
          className="button button--small form-summary__edit"
          onClick={[Function]}
          type="button"
        >
          Edit
        </button>
      </div>
    </div>
  </div>
</div>
`;
